src/vector_utils.cpp

```diff
@@ -3,6 +3,8 @@
 #include <numeric>    // std::accumulate
 #include <sstream>    // std::stringstream
 #include <string>     // std::string
+#include <cctype>     // std::tolower
+#include <cmath>
```

Comment:
They are not used anywhere. Why adding them?

Reply:
cctype is used for tolower, I don't remember for cmath but I'm pretty sure it is used by windows.

Reply:
if it's windows only, please wrap it in an ifdef, so it's obvious why it's needed.  Also, adding the comments like I did for what is used where is really helpful in the future.